.github/workflows/check-artifacthub-tags.yml

```diff
@@ -0,0 +1,92 @@
+# Check if all gadgets have tagged official and cncf in artifacthub.
+name: Check ArtifactHub Tags of Gadgets
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '0 1 * * *'
+
+env:
+  NO_COLOR: '\033[0m'
+  RED: '\033[0;31m'
+  GREEN: '\033[0;32m'
+  LIGHT_YELLOW: '\033[0;33m'
+
+jobs:
+  check-gadgets:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Gather valid gadgets
+      id: filter-gadgets
+      run: |
+        gadgets_dir="./gadgets"
+        valid_gadgets=()
+
+        # Loop to collect valid gadgets for check
+        for subdir in "$gadgets_dir"/*/; do
+          if [[ -f "$subdir/artifacthub-pkg.yml" ]]; then
+            subdir_name=$(basename "$subdir")
+            # Replace "_" with "-"
+            subdir_name=${subdir_name//_/-}
+            valid_gadgets+=("$subdir_name")
+          fi
+        done
+
+        valid_gadgets_string=$(IFS=,; echo "${valid_gadgets[*]}")
+        echo "valid-gadgets-string=${valid_gadgets_string}" >> $GITHUB_OUTPUT
+
+    - name: Check artifacthub tags
+      id: check-artifacthub-tags
+      env:
+        VALID_GADGETS_STRING: ${{ steps.filter-gadgets.outputs.valid-gadgets-string }}
+      run: |
+        passed_gadgets=()
+        failed_gadgets=()
+
+        # Iterate over each gadget name in the string
+        IFS=',' read -ra GADGETS <<< "$VALID_GADGETS_STRING"
+
+        if [ ${#GADGETS[@]} -eq 0 ]; then
+          echo -e "${LIGHT_YELLOW}Zero gadgets to be checked${NO_COLOR}"
+          exit 0
+        fi
+
+        for gadget in "${GADGETS[@]}"; do
+          echo -e "Checking gadget: ${LIGHT_YELLOW}$gadget${NO_COLOR}"
+
+          # Perform the curl request to get JSON data
+          response=$(curl -s -X 'GET' \
+            "https://artifacthub.io/api/v1/packages/inspektor-gadget/gadgets/$gadget" \
+            -H 'accept: application/json')
+
+          # Extract 'official' and 'cncf' values (if there are no keys, jq will return null)
+          official=$(echo "$response" | jq -r '.official')
+          cncf=$(echo "$response" | jq -r '.cncf')
+
+          if [[ "$official" == "true" && "$cncf" == "true" ]]; then
+            passed_gadgets+=("$gadget")
+          else
+            failed_gadgets+=("$gadget")
+          fi
+
+          # Gracefully handle rate limits by adding a 0.5-second delay between requests
+          sleep 0.5
+        done
+
+        if [ ${#passed_gadgets[@]} -gt 0 ]; then
+          echo -e "\n${GREEN}Passed Gadgets: ${passed_gadgets[@]}${NO_COLOR}"
+        else
+          echo -e "\n${RED}All Failed!${NO_COLOR}"
```

Comment:
Hi @eiffel-fl , thanks for the review !

Yeah we can, but we are handling this in below check. If we add `exit 1` here, then we will miss out printing failed gadgets (from below condition check).

--
Also on a side note, regarding the above comment on failing job.
Yes, the check was little uncertain. Actually I added a hardcoded value `herewego` into failed_gadgets arr to test condition execution and exiting status.

new failed [testcase](https://github.com/4rivappa/inspektor-gadget/actions/runs/11918644086/job/33216543259)
new passed [testcase](https://github.com/4rivappa/inspektor-gadget/actions/runs/11918563531/job/33216278702)

Maybe you can have a look at workflow file at that point of time, hope this would help to analyse testcase.

Reply:
yeah ok, got it.
After printing out all gadgets, we can `exit 1` from here.

Also I have added redirect to stderr from this point.
[Test workflow-run](https://github.com/4rivappa/inspektor-gadget/actions/runs/11939468983/job/33280051546)